daml-model/daml/Tests/Onboarding.daml

```diff
@@ -53,62 +53,69 @@ data GuiBackendConfigEntry = GuiBackendConfigEntry with
     name : Text
   deriving (Eq, Show)
 
-
 onboarding : [PartyInfo] -> Script Output
 onboarding partyInfo = do
   scheduler <- allocatePartyOn "Scheduler" (ParticipantName "schedulerParticipant")
   assembler <- allocatePartyOn "Assembler" (ParticipantName "assemblerParticipant")
-
-  (_, entries) <- foldlA
-    (\(parties, entries) info -> do
-      account <- setupAccount scheduler assembler parties info
-      let updatedParties = TextMap.insert info.name account.damlPartyId parties
-      pure (updatedParties, account :: entries))
-    (TextMap.empty, [])
-    partyInfo
-
-  pure $ Output { guiBackendConfig = entries, parties = fmap (.damlPartyId) entries }
-
+  parties <- setupParties partyInfo
+  entries <- mapA (\info -> setupAccount scheduler assembler parties info) partyInfo
+  pure $ Output { guiBackendConfig = entries, parties = map damlPartyId entries }
+
+setupParties : [PartyInfo] -> Script (TextMap Party)
+setupParties partyInfo = do
+  parties <- mapA (\info -> do 
+      debug $ "Creating party: " <> info.name
+      allocatePartyOn info.name (ParticipantName info.participantName)
+    ) partyInfo
+  pure $ TextMap.fromList $ zipWith (\info party -> (info.name, party)) partyInfo parties
 
 setupAccount : Party -> Party -> TextMap Party -> PartyInfo -> Script GuiBackendConfigEntry
 setupAccount scheduler assembler parties partyInfo = do
-  party <- allocatePartyOn partyInfo.name (ParticipantName partyInfo.participantName)
-
-  whenSome partyInfo.bic \bic ->
-    void $ party `submit` createCmd
-      BankBIC
-        with
-          bic = bic
-          banks = Set.singleton party
-          schedulers = Set.singleton scheduler
-          assemblers = Set.singleton assembler
-
-  whenSome partyInfo.account \account ->
+  let party = fromSomeNote
+        ("No party found for: " <> partyInfo.name)
+        (TextMap.lookup partyInfo.name parties)
+
+  whenSome partyInfo.bic \bic -> do
+    debug $ "Creating BankBic " <> bic <> " for " <> partyInfo.name
+    void $ party `submit` createCmd BankBIC
+      with
+        bic = bic
+        bankShards = Set.singleton party
+        schedulerShards = Set.singleton scheduler
+        assemblerShards = Set.singleton assembler
+        otherBanks = Set.fromList . textMapValues $ TextMap.filter (/= party) parties
```

Comment:
I think this does not belong here. I'd rather move this out to a dedicated function called from `onboarding`